src/Security/Member.php

```diff
@@ -953,17 +954,16 @@ public function onBeforeWrite()
 
         // We don't send emails out on dev/tests sites to prevent accidentally spamming users.
         // However, if TestMailer is in use this isn't a risk.
-        if ((Director::isLive() || Email::mailer() instanceof TestMailer)
+        if ((Director::isLive() || Mailer::get_inst() instanceof TestMailer)
```

Comment:
I'd like to get rid of these references completely. Are tests not run under "live" mode?